### DISCUSSION TITLE
R scripts for MDA analysis?

### DISCUSSION THREAD

**Discussion by MinaAlmasi** [mshakirDr/MFTE#12] [Q&A] [answered]
Hi! Thank you for your great work!

I'm looking into maybe using your library MFTE to extract features for a large range of texts (citing your software of course!), and I was wondering how far you were with adding the R scripts for MDA analysis that you mention in the README? 

Best
Mina

**Comment by mshakirDr** [comment, marked as answer]
Hi
Thanks for using our software. The addition of R scripts plan has not happened yet. Doing MD analysis in R is not difficult. There are a ton of online resources and tutorials available which can be helpful. 
For what it's worth, I have attached a zip file containing an R Markdown. It may be helpful for MD analysis (Exploratory Factor Analysis). The code is poorly written (for personal use) and not very well documented/commented. Please use it if you know what you are doing. I cannot guarantee if it will run on your system without error. Again, please use only if you are familiar with R, and know how to debug/change such code.
[MD_analysis_sample.zip](https://github.com/user-attachments/files/17779302/MD_analysis_sample.zip)

**Reply by MinaAlmasi** [reply]
Hi! 

Sorry for getting back so late - I did not receive a notification for some reason! Thank you for your answer and for the code. 

I am familiar with R - I was also curious about your approach to doing it. My impression is that there are some methodological choices that can affect the outcome of Exploratory Factor Analysis (but correct me if I'm wrong - I have only done PCA). 

It was not to stress you about adding the R code - I was purely curious. Thanks again for your response and code!

**Reply by mshakirDr** [reply]
Hi
I am not an expert on the use of PCA or EFA. But there is surely good literature available within corpus linguistics that explains these methods. There is one tutorial paper by Egbert and Staples (2019). http://dx.doi.org/10.5040/9781350023857.0015
Stephanie Evert has written on the use of dimensionality reduction methods (including PCA). You can find her website and access many materials. 
@elenlefoll 's book (based on her PhD thesis) has good discussion on the use of different methods (EFA, PCA etc.).
The RMD file is also based on what I have learnt from these people. I hope the same will be helpful for you as well.

**Reply by elenlefoll** [reply]
Hi @MinaAlmasi!
Thanks for your questions.
I have published all of the analysis scripts that I wrote to conduct my research that has used (an earlier version of) the MFTE. Perhaps some of those are helpful:
- https://elenlefoll.github.io/TextbookMDA/
- https://osf.io/n3fz9/
In both the book and the article mentioned above, I also discuss the most important differences between PCA and EFA in the context of MDA.
In addition, Shakir and I wrote paper on the MFTE that has just been published in Research in Corpus Linguistics: [https://doi.org/10.32714/ricl.13.02.03](http://ricl.aelinco.es/first-view/358-Article%20Text-2960-1-10-20241130.pdf). If you use the MFTE in your research, please cite this paper.
Many thanks and best wishes

**Reply by MinaAlmasi** [reply]
Thank you both for all the ressources! Really appreciate you taking your time to write detailed responses like this. 

We will definitely make sure your work is cited if we use it!
